index.bs

```diff
@@ -6014,6 +6061,85 @@ To <dfn data-lt="bidirectionally broadcasting the shapes">bidirectionally broadc
+        : {{MLOperandDataType/"int64"}}
+        :: Return [=ConvertToInt=](|x|, 64, "signed").
+        : {{MLOperandDataType/"uint64"}}
+        :: Return [=ConvertToInt=](|x|, 64, "unsigned").
+        : {{MLOperandDataType/"int32"}}
+        :: Return [=ConvertToInt=](|x|, 32, "signed").
+        : {{MLOperandDataType/"uint32"}}
+        :: Return [=ConvertToInt=](|x|, 32, "signed").
+        : {{MLOperandDataType/"int8"}}
+        :: Return [=ConvertToInt=](|x|, 8, "signed").
+        : {{MLOperandDataType/"uint8"}}
+        :: Return [=ConvertToInt=](|x|, 8, "unsigned").
+    </dl>
+
+NOTE: The input to [=cast=] is an abstract number with unlimited range and precision, including the special values Infinity, -Infinity and NaN. The output is also an abstract number, but exactly representable as the specified type.
+
+<div algorithm>
+The <dfn>ConvertToFloat</dfn>(|x|, |bitLength|) steps are:
+
+1. If |x| is NaN, then return NaN.
+1. Switch on |bitLength|:
+    <dl class=switch>
+        : 32
+        ::
+            1. Let |upperBound| be 2<sup>128</sup>.
+            1. Let |lowerBound| be -2<sup>128</sup>.
+            1. Let |S| be the set of [[IEEE-754-2019]] binary32 floating point values except -0, but with the special values |upperBound| and |lowerBound| added.
+        : 16
+        ::
+            1. Let |upperBound| be 2<sup>16</sup>.
+            1. Let |lowerBound| be -2<sup>16</sup>.
+            1. Let |S| be the set of [[IEEE-754-2019]] binary16 floating point values except -0, but with the special values |upperBound| and |lowerBound| added.
+    </dl>
+1. Let |y| be the number in |S| that is closest to |x|, selecting the number with an even significand if there are two [=equally close values=]. The two special values |lowerBound| and |upperBound| are considered to have even significands for this purpose.
+1. If |y| is |upperBound|, then return +Infinity.
+1. If |y| is |lowerBound|, then return -Infinity.
+1. If |y| is +0 and |x| is negative, return -0.
+1. Return |y|.
+
+NOTE: This is based on a definition in [[WEBIDL]], but extended to cover 16-bit floating point values.
+
+</div>
+
+<div algorithm>
+The <dfn>ConvertToInt</dfn>(|x|, |bitLength|, |signedness|) steps are:
+
+1. If |signedness| is "unsigned", then:
+    1. Let |lowerBound| be 0.
+    1. Let |upperBound| be 2<sup>|bitLength|</sup> - 1.
+1. Otherwise:
+    1. Let |lowerBound| be -2<sup>|bitLength| - 1</sup>.
+    1. Let |upperBound| be 2<sup>|bitLength| - 1</sup> - 1.
+1. If |x| is -0, then set |x| to +0.
+1. If |x| is NaN, then return +0.
+1. Set |x| to min(max(|x|, |lowerBound|), |upperBound|).
+1. Round |x| to the nearest integer, choosing the even integer if it lies halfway between two, and choosing +0 rather than -0.
+1. Return |x|.
+
+NOTE: This is based on a definition in [[WEBIDL]] with these differences: 64-bit integers are not treated specially, the input |x| is an abstract number, and clamping is always performed.
```

Comment:
I can see clamping resulting in unexpected behavior from the developer's point of view, though the alternative of throwing an exception may not be desirable either ¯\\_(ツ)_/¯

Seems reasonable to start with clamping for now

Reply:
Yeah, when I was initially crafting this I was thinking we'd want [EnforceRange](https://webidl.spec.whatwg.org/#EnforceRange) behavior - like we do for integer arguments - but that gets real weird real fast since doing the checks sometimes happens at activation fusion time, and the same code and constants might suddenly throw if you switch from using float32 to float16.

The places in the spec that still take integers and use EnforceRange still make sense, since you'd never want clamping (or worse, wrapping) behavior when specifying an axis, dimension, etc.